src/ray/raylet/lineage_cache.cc

```diff
@@ -23,6 +23,14 @@ void LineageEntry::ResetStatus(GcsStatus new_status) {
   status_ = new_status;
 }
 
+void LineageEntry::MarkExplicitlyForwarded(const ClientID &node_id) {
+  forwarded_to_.insert(node_id);
+}
+
+bool LineageEntry::GetExplicitForwardingStatus(const ClientID &node_id) const {
```

Comment:
Maybe better to rename to ```IsExplicitlyForwarded```?

A related question is: what's the difference between _explicitly forwarded_ and _implicitly forwarded_?  It looks like that _implicitly forwarded_ simply indicates _have been explicitly forwarded_? If that's true maybe we could just use _forwarded_?

Reply:
Yeah not too sure on how to refer to it, but I wanted to distinguish between the node manager having explicitly marked a task as `forwarded`, and the fact that all ancestors of this task have also been forwarded, but haven't been marked as such (for efficiency).

For example assume `T1 -> T2 -> T3` is uncommitted lineage. `T1` is first forwarded from `n1` to `n2` (and marked). Then on a later call, `T2 -> T3` is forwarded to `n2` but **only** `T3` is marked to prevent having to go through the entire lineage till the previously last forwarded.

That's why I think referring to the state as just `forwarded`  is a bit misleading, since for example, `t2.isForwarded(n2)`.

Open to suggestions on how to make this more clear (or if this is incorrect behaviour).

Reply:
Makes sense, thanks.  Remote call arguments should belong to ```implicit``` tasks, as they are always forwarded along the task of the remote call. Do we have other scenarios that use ```implicit``` tasks?

Reply:
Remote call arguments won't necessarily belong to `implicit` tasks. I think @ujvl's point is that a task will be implicitly forwarded if it was sent as part of a task's uncommitted lineage, and explicitly forwarded if the task itself is the one being forwarded.